docs/source/upgrade-2.0.rst

```diff
@@ -821,9 +821,240 @@ Rationale
 These features were removed to better align cocotb's :class:`!Event` with :class:`asyncio.Event`.
 
 
-*********************************************************
-Use :func:`!cocotb.start_soon` over :func:`!cocotb.start`
-*********************************************************
+****************************************************
+Remove uses of :class:`!Join` and :meth:`!Task.join`
+****************************************************
+
+Change
+======
+
+:class:`.Join` and :meth:`.Task.join` have been deprecated.
+
+How to Upgrade
+==============
+
+Remove :class:`!Join` and :meth:`!Task.join` and simply :keyword:`await` the :class:`!Task` being joined,
+or pass the :class:`!Task` directly into the function that will :keyword:`await` it.
+
+.. code-block:: python
+    :caption: Example Task
+
+    async def example():
+        ...
+
+    task = cocotb.start_soon()
+
+.. code-block:: python
+    :caption: Old way using :class:`!Join`
+    :class: removed
+
+    await Join(task)
+    # or
+    await First(Join(task), RisingEdge(cocotb.top.clk))
+
+.. code-block:: python
+    :caption: Old way using :meth:`!Task.join`
+    :class: removed
+
+    await task.join()
+    # or
+    await First(task.join(), RisingEdge(cocotb.top.clk))
+
+.. code-block:: python
+    :caption: New way using :class:`!Task` directly
+    :class: new
+
+    await task
+    # or
+    await First(task, RisingEdge(cocotb.top.clk))
+
+Rationale
+=========
+
+There were multiple synonyms for the same functionality, so we deprecated all but the least verbose.
+This also better aligns cocotb's :class:`!Task`\ s with :class:`asyncio.Task`.
+
+Additional Details
+==================
+
+:keyword:`await`\ ing a :class:`!Join` trigger would return the result of the :class:`!Task`, not the trigger.
+This made determining what trigger fired difficult since there is no way to determine which :class:`!Task` returned a value if they all return the same type.
+:class:`.TaskComplete` was added to solve this deficiency in :class:`!Join`.
+Use :attr:`.Task.complete` to get the :class:`!TaskComplete` trigger associated with each :class:`!Task`.
```

Comment:
Waiting on a `Task` still has the same problem.
```suggestion
:keyword:`await`\ ing a :class:`!Task` (or previously a :class:`!Join` trigger) returns the result of the :class:`!Task`, not the task/trigger.
This can make it difficult to determine which :class:`!Task: finished when waiting on multiple, such as with a :class:`.First` trigger.
:class:`.TaskComplete` was added to solve this.
Use :attr:`.Task.complete` to get the :class:`!TaskComplete` trigger associated with each :class:`!Task`.
```